dbms/src/main/java/org/frankframework/dbms/MariaDbDbmsSupport.java

```diff
@@ -62,7 +57,10 @@ public boolean hasSkipLockedFunctionality() {
 
 	private boolean determineSkipLockedCapability(String productVersion) {
 		String[] productVersionArr = productVersion.split("-");
-		String strippedProductVersion = productVersionArr.length > 1 ? productVersionArr[1] : productVersion;
+		// The part of productVersion to use depends on whether the MariaDB or MySQL driver is used.
+		// MySQL driver prepends its own version and so we have to take the 2nd entry in the array.
+		// When MariaDB driver is used, take the 1st entry.
+		String strippedProductVersion = productVersionArr.length == 1 || productVersionArr[1].toLowerCase().contains("maria") ? productVersionArr[0] : productVersionArr[1];
```

Comment:
Voor de leesbaarheid zou ik dit stukje (evt vanaf regel 59) extracten naar een aparte method `getStrippedProductVersion` oid.

Reply:
Deze methode is al apart, zijnde uniek voor deze implementatie, en is verder niet zo groot. Ik snap wel dat deze one-liner een beetje extreem is. Kan je niet string contains doen?